internal/orchestration/worker.go

```diff
@@ -163,25 +163,24 @@ func (w *Worker) createReplica(opts *orchestrationpb.ReplicaOpts) (*replica.Repl
 	// prepare modules
 	builder := consensus.NewBuilder(hotstuff.ID(opts.GetID()), privKey)
 
-	var consensusRules consensus.Rules
-	if !modules.GetModule(opts.GetConsensus(), &consensusRules) {
+	consensusRules, ok := modules.GetModule[consensus.Rules](opts.GetConsensus())
+	if !ok {
 		return nil, fmt.Errorf("invalid consensus name: '%s'", opts.GetConsensus())
 	}
 
-	var byz byzantine.Byzantine
-	if modules.GetModule(opts.GetByzantineStrategy(), &byz) {
+	if byz, ok := modules.GetModule[byzantine.Byzantine](opts.GetByzantineStrategy()); ok {
 		consensusRules = byz.Wrap(consensusRules)
 	} else if opts.GetByzantineStrategy() != "" {
```

Comment:
Don't you want to err out `if !ok`? If I understand this, it means an unknown/invalid byzantine strategy was supplied.

Reply:
This is indeed a bit confusing. I have refactored it. What really should happen is this:

1. If `ByzantineStrategy` is empty, do nothing
1. If `GetModule` succeeds, wrap the consensus rules
1. Else return an error